### PR TITLE
Hib 142 create participants api for sponsor

### PR DESCRIPTION
**Pull request by hwangseung**
### Description

(_What have been done/changed_)

- Added endpoints for getting and updating a specific participant.

### Testing plan

- Tested manually with postman.

### CODE REVIEW

File: supabase/migrations/20230124013902_add_participants.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+alter table "public"."participants" enable row level security;
+
+CREATE UNIQUE INDEX participants_pkey ON public.participants USING btree (id);
+
+alter table "public"."participants" add constraint "participants_pkey" PRIMARY KEY using index "participants_pkey";
+
+alter table "public"."participants" add constraint "participants_id_fkey" FOREIGN KEY (id) REFERENCES user_profiles(user_id) not valid;
+
+alter table "public"."participants" validate constraint "participants_id_fkey";
+
```
Comment (snokpok) [inline supabase/migrations/20230124013902_add_participants.sql:24]: this migration conflicts with those you merged earlier and was deployed to prod in `main`; I just merged `main` into `dev` to have that migration so you can update this branch with `dev`; you can delete this

File: supabase/migrations/20230125031217_add_notes_and_edit_participants.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+CREATE UNIQUE INDEX notes_pkey ON public.notes USING btree (id);
+
+alter table "public"."notes" add constraint "notes_pkey" PRIMARY KEY using index "notes_pkey";
+
+alter table "public"."notes" add constraint "notes_participant_id_fkey" FOREIGN KEY (participant_id) REFERENCES participants(id) not valid;
+
+alter table "public"."notes" validate constraint "notes_participant_id_fkey";
+
+
```
Comment (snokpok) [inline supabase/migrations/20230125031217_add_notes_and_edit_participants.sql:20]: same comment with above

File: apps/dashboard/pages/api/participants/[participantId].ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import { PostgrestError } from '@supabase/supabase-js';
+import e from 'express';
+
+/**
+ * Creates a team in the teams table
+ * @param req - (name, description, photoKey, userId) : (string, string,
+ *      string-link to photo, string-user_id of person creating team)
+ * @param res - Only used for returning Json messages
+ * @return 500 if error in creation. 200 if team created successfully
+ */
```
Comment (snokpok) [inline apps/dashboard/pages/api/participants/[participantId].ts:14]: invalid docstring

**Review by snokpok — CHANGES_REQUESTED** [review]
some changes

**Comment by hwangseung** [comment]
Changes made.

**Pull request closed without merging**